src/matcher.rs

```diff
@@ -0,0 +1,160 @@
+/// Check if input is a valid domain label
+pub fn is_label(input: &str) -> bool {
+    if input.is_empty() {
+        return false;
+    }
+    let mut chars = input.chars();
+
+    // it can start with an alphanumeric character
+    let first = chars.next().unwrap();
+    if !first.is_ascii_alphanumeric() {
+        return false;
```

Comment:
You can eliminate `unwrap` by using `match` for example like below. That `unwrap` should never panic because the length is checked above, but it's also possible to implement same thing without it, so why not?

```suggestion
    match chars.next() {
        Some(c) if c.is_ascii_alphanumeric() => (),
        _ => return false,
    }
```

Reply:
Ok!

Reply:
I've replaced the `unwrap` with match which removes the need for checking for `is_empty()`